inference/enterprise/workflows/core_steps/transformations/detection_filter.py

```diff
@@ -0,0 +1,215 @@
+    type: Literal["CompoundDetectionFilterDefinition"]
+    left: Annotated[
+        Union[DetectionFilterDefinition, "CompoundDetectionFilterDefinition"],
+        Field(
+            discriminator="type",
+            description="Left operand (potentially nested expression) in expression `left bin_operator right`",
+        ),
+    ]
+    operator: BinaryOperator = Field(
+        description="Binary operator in expression `left bin_operator right`",
+        examples=["and", "or"],
+    )
+    right: Annotated[
+        Union[DetectionFilterDefinition, "CompoundDetectionFilterDefinition"],
+        Field(
+            discriminator="type",
+            description="Right operand (potentially nested expression) in expression `left bin_operator right`",
+        ),
+    ]
+
+
+LONG_DESCRIPTION = """
+Filter detections from a detections-based block based on specified conditions.
+
+This block is useful to filter out detections that may not be useful for your project. 
+You can filter on the basis of:
+
+- Coordinates of a detection
+- Confidence of a detection
+
+You can use the following comparison statements in a DetectionFilterBlock:
+
+- `equal` (field value equal to `reference_value`)
+- `not_equal`
+- `lower_than`
+- `greater_than`
+- `lower_or_equal_than`
+- `greater_or_equal_than`
+- `in`
+- `str_starts_with`
+- `str_ends_with`
+- `str_contains`
+"""
+
+SHORT_DESCRIPTION = (
+    "Filter predictions from detection models based on defined " "conditions."
+)
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "transformation",
+        }
+    )
+    type: Literal["DetectionFilter"]
+    predictions: StepOutputSelector(
+        kind=[
+            OBJECT_DETECTION_PREDICTION_KIND,
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            KEYPOINT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Reference to detection-like predictions",
+        examples=["$steps.object_detection_model.predictions"],
+    )
+    filter_definition: Annotated[
+        Union[DetectionFilterDefinition, CompoundDetectionFilterDefinition],
+        Field(
+            discriminator="type",
+            description="Definition of a filter expression to be applied for each element of detection-like predictions to decide if element should persist in the output. Can be used for instance to filter-out bounding boxes based on coordinates.",
+        ),
+    ]
+    image_metadata: Annotated[
+        StepOutputSelector(kind=[IMAGE_METADATA_KIND]),
+        Field(
+            description="Metadata of image used to create `predictions`. Must be output from the step referred in `predictions` field",
+            examples=["$steps.detection.image"],
+        ),
+    ]
+    prediction_type: Annotated[
```

Comment:
When we combine the internal data for predictions, does both `image_metadata` and `prediction_type` become part of `predictions` ?

Reply:
yes! that's one of the painpoint and source of design-induced complexity now.
When we change into sv.Detections(...) - we have all of that + information about parent embeded.